webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -871,6 +875,45 @@ protected function validateTeam(array $event): void
         $this->processPendingEvents('team', $team->getTeamid());
     }
 
+    protected function validateAccount(array $event): void
+    {
+        $userId = $event['data']['id'];
+
+        /** @var User|null $user */
+        $user = $this->em
+            ->getRepository(User::class)
+            ->findOneBy(['externalid' => $userId]);
+
+        if ($event['op'] === EventLogService::ACTION_DELETE) {
+            // We need to check if the user is known
+            if ($user) {
+                $this->addOrUpdateWarning($event, ExternalSourceWarning::TYPE_ENTITY_SHOULD_NOT_EXIST);
+            }
+
+            $this->removeWarning($event, ExternalSourceWarning::TYPE_ENTITY_SHOULD_NOT_EXIST);
```

Comment:
Why is the warning directly removed?

Reply:
I don’t follow.  We remove the warning that the user should not exist if it is stil there but the user is deleted externally since we are in sync again. But it misses a return two lines above. Which is copy pasted so it misses somewhere else as well

Reply:
That explains why you directly remove the Warning from 890.